Sources/CMakeLists.txt

```diff
@@ -6,28 +6,100 @@
 # See http://swift.org/LICENSE.txt for license information
 # See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 
-# Macros must be built for the build machine, not the host.
-include(ExternalProject)
-if(NOT SwiftTesting_MACRO_MAKE_PROGRAM)
-  set(SwiftTesting_MACRO_MAKE_PROGRAM ${CMAKE_MAKE_PROGRAM})
-endif()
+set(SwiftTesting_MACRO "<auto>" CACHE STRING
+    "Path to SwiftTesting macro plugin, or '<auto>' for automatically building it")
+
+if(SwiftTesting_MACRO STREQUAL "<auto>")
+  # Macros must be built for the build machine, not the host.
+  include(ExternalProject)
+  if(NOT SwiftTesting_MACRO_MAKE_PROGRAM)
+    set(SwiftTesting_MACRO_MAKE_PROGRAM ${CMAKE_MAKE_PROGRAM})
+  endif()
+  if(NOT SwiftTesting_MACRO_Swift_COMPILER)
+    set(SwiftTesting_MACRO_Swift_COMPILER ${CMAKE_Swift_COMPILER})
+  endif()
+  if(NOT SwiftTesting_MACRO_Swift_FLAGS)
+    set(SwiftTesting_MACRO_Swift_FLAGS ${CMAKE_Swift_FLAGS})
```

Comment:
What flags are you hoping to pull here?
This seems risky given that these flags might be for an entirely different machine than the one the macros will build for/run on.

Reply:
This is actually a leftover from when I was trying around. Let me remove this as a followup.